### PR TITLE
[tegnamedia] Add new information extractors

### PR DESCRIPTION
**Pull request by goggle**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description
As requested in #12153 I have added information extractors for three TEGNA Media sites, including 9news.com. 

I did not manage to extract the HDS format, can somebody have a look at this? [Here](http://api.tegna-tv.com/video/v2/getAllVideoPathsById/1073002512310/73?subscription-key=ae1d3e46c9914e9b87757fead91d7654) is a link to a json, which contains the relevant information.

There are many Tegna news sites (see [here](http://www.tegna.com/media/)), which use the exact same scheme. Should I add support for all of them, like I did with 12news.com and thv11.com? If so, should I add a test for all of them, like I did for 12news.com?

### CODE REVIEW

**Comment by remitamine** [comment]
> I did not manage to extract the HDS format, can somebody have a look at this?

http://kusa-vh.akamaihd.net/z/video/2512310/2512310_,350,640,1000,2000,.mp4.csmil -> http://kusa-vh.akamaihd.net/z/video/2512310/2512310_,350,640,1000,2000,.mp4.csmil/manifest.f4m?hdcore

**Comment by goggle** [comment]
@remitamine Thanks, I've added support for HDS to the code.

File: youtube_dl/extractor/tegnamedia.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            'title': 'Father worries about immigration status',
+            'description': '9NEWS @ 9. 2/15/2017',
+            'thumbnail': 'http://kusa-download.edgesuite.net/video/2512310/2512310_Still.jpg',
+            'duration': 96.0,
+            'timestamp': 1487218434,
+            'upload_date': '20170216',
+        }
+    }
+
+    def _real_extract(self, url):
```
Comment (dstftw) [inline youtube_dl/extractor/tegnamedia.py]: There is no need in this method.

File: youtube_dl/extractor/tegnamedia.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            'description': 'md5:24188e754669c29700e8dd6d19e4943b',
+            'timestamp': 1487360943,
+            'upload_date': '20170217',
+        },
+        'params': {
+            'skip_download': True,
+        }
+    }
+
+    def _real_extract(self, url):
```
Comment (dstftw) [inline youtube_dl/extractor/tegnamedia.py]: There is no need in this method.

File: youtube_dl/extractor/tegnamedia.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        return super(TwelveNewsIE, self)._real_extract(url)
+
+
+class THVElevenIE(TegnaMediaIE):
```
Comment (dstftw) [inline youtube_dl/extractor/tegnamedia.py]: There is no need in this method.

File: youtube_dl/extractor/tegnamedia.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            'description': description,
+            'thumbnail': thumbnail,
+            'duration': duration,
+            'timestamp': timestamp,
+            'formats': formats,
+        }
+
+
+class NineNewsIE(TegnaMediaIE):
+    _VALID_URL = r'https?://(?:www\.)?9news\.com/.+/(?P<id>[0-9]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/tegnamedia.py]: This allows URLs like http://www.9news.com/////408808900 that is incorrect.

File: youtube_dl/extractor/tegnamedia.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            r'data-id="(?P<id>\d+)"', player_info, 'video id')
+        data_site = self._search_regex(
+            r'data-site="(?P<data_site>\d+)"', player_info, 'data site')
+
+        api_url = 'http://api.tegna-tv.com/video/v2/getAllVideoPathsById/%s/%s?subscription-key=%s' % (data_id, data_site, self.SUBSCRIPTION_KEY)
+        video_json = self._download_json(api_url, show_id)
+
+        video_id = str_or_none(video_json['Id'])
+        title = str_or_none(video_json['Title'])
+        description = str_or_none(video_json['Description'])
```
Comment (dstftw) [inline youtube_dl/extractor/tegnamedia.py]: None of the optional metadata should break the extraction.

File: youtube_dl/extractor/tegnamedia.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                formats.append(
+                    {
+                        'url': path,
+                        'format_id': 'mp4-' + str_or_none(elem['EncodingRate']),
+                        'vbr': int_or_none(elem['EncodingRate']),
+                    }
+                )
+            elif elem.get('Format') == 'HLS':
+                forms = self._extract_m3u8_formats(
+                    path, video_id, ext='mp4', entry_protocol='m3u8_native')
```
Comment (dstftw) [inline youtube_dl/extractor/tegnamedia.py]: Should not be fatal.

**Pull request closed without merging**